iroh-net-report/src/reportgen/probes.rs

```diff
@@ -261,19 +265,30 @@ impl ProbePlan {
                     })
                     .expect("adding QuicIpv6 probe to a QuicAddrIpv6 probe set");
             }
-            plan.add(stun_ipv4_probes);
-            plan.add(stun_ipv6_probes);
-            plan.add(quic_ipv4_probes);
-            plan.add(quic_ipv6_probes);
+            plan.add_probes(
+                &protocols,
+                vec![
+                    stun_ipv4_probes,
+                    stun_ipv6_probes,
+                    quic_ipv4_probes,
+                    quic_ipv6_probes,
+                ],
+            );
 
             // The HTTP and ICMP probes only start after the STUN probes have had a chance.
             let mut https_probes = ProbeSet::new(ProbeProto::Https);
             let mut icmp_probes_ipv4 = ProbeSet::new(ProbeProto::IcmpV4);
             let mut icmp_probes_ipv6 = ProbeSet::new(ProbeProto::IcmpV6);
+
+            let has_priority_probes = plan.has_priority_probes();
             for attempt in 0..3 {
-                let start = *max_stun_delay.get_or_insert_with(|| plan.max_delay())
-                    + DEFAULT_INITIAL_RETRANSMIT;
-                let delay = start + DEFAULT_INITIAL_RETRANSMIT * attempt as u32;
+                let delay = if !has_priority_probes {
```

Comment:
I'm not entirely sure why this needs changing.  The old version uses `let start = *max_stun_delay.get_or_insert_with(|| plan.max_delay())`.  I would expect `plan.max_delay()` to return 0 if there were no high-prio probes.  So then `start` should also be 0?

Admittedly I'm just reading code and didn't try this out.

Reply:
okay, I definitely changed this in the most confusing way, so refactored and added comments.

Basically, if there are no high priority probes, then we don't need `start` to also add the `DEFAULT_INITIAL_RETRANSMIT`, but if it DOES than we want to add that buffer.

Hopefully this is clearer. The issue was that if you don't have any high prio probes in the list of probes you are attempting, then all the probes added in this section started with a 100ms delay.